cmake/cfe-common.doxyfile.in

```diff
@@ -77,6 +75,21 @@ INPUT                 += @MISSION_SOURCE_DIR@/cfe/docs/src/cfe_xref.dox
 FILE_PATTERNS          = *.c *.cpp *.cc *.C *.h *.hh *.hpp *.H *.dox *.md
 RECURSIVE              = YES
 EXAMPLE_PATTERNS       = *
+
+# Exclude coverage test stubs, examples, and test framework from the documentation
+EXCLUDE_PATTERNS       += "*/ut-stubs/*"
+EXCLUDE_PATTERNS       += "*/ut-coverage/*"
+EXCLUDE_PATTERNS       += "*/examples/*"
+EXCLUDE_PATTERNS       += "*/tests/*"
+EXCLUDE_PATTERNS       += "*/unit-test*/*"
+
+# Exclude UT assert itself (part of OSAL)
+EXCLUDE                += @MISSION_SOURCE_DIR@/osal/ut_assert
+
+# Also exclude the OSAL main documentation from the cFE doc build
+# (this may create a conflict by having more than one main page)
+EXCLUDE                += @MISSION_SOURCE_DIR@/osal/doc
```

Comment:
Could these exclusions be owned by osal assuming it has it's own doxyfile that gets included when the detaildesign is built?  I'd only expect cfe exclusions here.

Reply:
The "osal/doc" exclusion really does need to be here, because it has a "mainpage" which conflicts with the CFE mainpage.  In other words, the exclusion is specific to the CFE doc build.

As for UT assert exclusion, I would actually prefer to remove this exclusion.  It hadn't been included in docs before, but now it does have some doxygen markup, and since it is part of the framework (esp when functional testing cFE via cfe_assert) it makes sense to include it.  

The only problem is that it was causing a bunch of extra warnings, so I thought I'd circle back and correct the warnings first, then perhaps we can remove this exclude line.